### ISSUE TITLE
The NVDA screen reader announces "sub-menu" when an object contains aria-haspopup="dialog".

### ISSUE THREAD

**Issue by accdc** [nvaccess/nvda#8254]
The NVDA screen reader announces "sub-menu" when an object contains aria-haspopup="dialog".

NVDA (2018.1.1) 
Firefox (v60, ESR)
Windows7, Windows8, Windows10.

```
//begin code snippet// 
<a href="#" role="button" aria-haspopup="dialog">I have a dialog</a> 
//end code snippet//
```

To recreate: Tab to the object that contains aria-haspopup="dialog" and verify that NVDA announces "sub-menu"

When aria-haspopup is set to "true" or "menu", then it is safe to be conveyed as having a menu attached, but all other tokens should be accurately conveyed based on their value assignment. E.G "dialog", "listbox", "tree", "grid", etc.

**Comment by leonardder** [comment]
Closing as duplicate of #8235.
